apiserver/facades/client/applicationoffers/base.go

```diff
@@ -227,14 +227,30 @@ func (api *BaseAPI) getOfferAdminDetails(user names.UserTag, backend Backend, ap
 // checkOfferAccess returns the level of access the authenticated user has to the offer,
 // so long as it is greater than the requested perm.
 func (api *BaseAPI) checkOfferAccess(user names.UserTag, backend Backend, offerUUID string) (permission.Access, error) {
-	access, err := backend.GetOfferAccess(offerUUID, user)
-	if err != nil && !errors.IsNotFound(err) {
+	err := api.Authorizer.EntityHasPermission(user, permission.SuperuserAccess, backend.ControllerTag())
+	if err != nil && !errors.Is(err, authentication.ErrorEntityMissingPermission) {
+		return permission.NoAccess, errors.Trace(err)
+	} else if err == nil {
+		return permission.AdminAccess, nil
+	}
+
+	err = api.Authorizer.EntityHasPermission(user, permission.AdminAccess, backend.ModelTag())
+	if err != nil && !errors.Is(err, authentication.ErrorEntityMissingPermission) {
 		return permission.NoAccess, errors.Trace(err)
+	} else if err == nil {
+		return permission.AdminAccess, nil
 	}
-	if !access.EqualOrGreaterOfferAccessThan(permission.ReadAccess) {
-		return permission.NoAccess, nil
+
+	for _, access := range []permission.Access{permission.AdminAccess, permission.ConsumeAccess, permission.ReadAccess} {
```

Comment:
Should we leave a comment that we specifically loop over the access levels in descending order in order to return the highest level of access.
It might be non-obvious coming back here after a while.

Reply:
added